appeals/web/src/server/lib/validators/date-input.validator.js

```diff
@@ -14,119 +14,69 @@ export const createDateInputFieldsValidator = (
 	messageFieldNamePrefix = 'date',
 	dayFieldName = '-day',
 	monthFieldName = '-month',
-	yearFieldName = '-year'
-) =>
-	createValidator(
-		body(`${fieldNamePrefix}${dayFieldName}`)
-			.exists()
-			.withMessage(
-				capitalize(
-					`${(messageFieldNamePrefix && messageFieldNamePrefix + ' d') || 'D'}ay cannot be empty`
-				)
-			)
-			.bail()
-			.trim()
-			.notEmpty()
-			.withMessage(
-				capitalize(
-					`${(messageFieldNamePrefix && messageFieldNamePrefix + ' d') || 'D'}ay cannot be empty`
-				)
-			)
-			.bail()
+	yearFieldName = '-year',
+	bodyScope = ''
+) => {
+	const _day = bodyScope ? `[${dayFieldName}]` : dayFieldName;
+	const _month = bodyScope ? `[${monthFieldName}]` : monthFieldName;
+	const _year = bodyScope ? `[${yearFieldName}]` : yearFieldName;
+
+	return createValidator(
+		body(bodyScope).custom((value) => {
+			const day = value[`${fieldNamePrefix}${dayFieldName}`];
+			const month = value[`${fieldNamePrefix}${monthFieldName}`];
+			const year = value[`${fieldNamePrefix}${yearFieldName}`];
+
+			if (day && month && year) {
+				return true;
+			}
+
+			let missingParts = [];
+			if (!day) missingParts.push('a day');
+			if (!month) missingParts.push('a month');
+			if (!year) missingParts.push('a year');
+
+			const messageSuffix = missingParts.reduce((acc, part, index) => {
```

Comment:
This is a great improvement, it's nice to see the snags with the existing validation being fixed! There is some fairly complex logic here, so just wondering if it might be worth adding unit tests for this? The main bits to test would probably be the bodyScope and the reducer that builds the error message. It might be that these are sufficiently covered by existing tests on feature code that uses the validator though, so feel free to ignore this comment if that's the case